admin-dev/themes/new-theme/.webpack/dev.js

```diff
@@ -11,10 +16,68 @@ function devConfig() {
     {
       devtool: 'inline-source-map',
       devServer: {
+        client: {
+          logging: 'error',
+          progress: false,
+          overlay: {
+            errors: true,
+            warnings: false,
+          },
+        },
         hot: true,
-        contentBase: path.resolve(__dirname, '/../public'),
-        publicPath: '/',
+        static: {
+          directory: path.join(__dirname, '/../public'),
+          watch: false,
+        },
+        port: 3000,
+        watchFiles: [
+          path.join(__dirname, '/../**/*.tpl'),
+          path.join(__dirname, '../js/**/*.js'),
+          path.join(__dirname, '../scss/**/*.scss'),
+        ],
+        open: true,
+        proxy: {
+          '**': {
+            target: process.env.PS_URL,
+            secure: false,
+            changeOrigin: true,
+          },
+        },
+        devMiddleware: {
+          publicPath: path.join(__dirname, '/../public'),
+          writeToDisk: (filePath) => !(/hot-update/.test(filePath)),
+        },
       },
+      plugins: [
+        new webpack.HotModuleReplacementPlugin(),
+        new HotAcceptPlugin({
+          test: [
+            ...Object.keys(common.entry).map((el) => `${common.entry[el]}.js`),
```

Comment:
So if I understand correctly this will return an array built on the `entries` from common file. So Something like `['manufacturer_address_form.js', 'merchandise_return.js', ...]`

Does it mean it will only watch for the "index" file of each component, or does it watch the bundle file? will it still trigger reload when the sub-components are modified as well? I don't understand how the matching is handled but maybe it's all internal and there is non need to worry about it? 🤷‍♂️